Objects/funcobject.c

```diff
@@ -9,6 +9,29 @@
 
 static uint32_t next_func_version = 1;
 
+static PyFunction_EventCallback func_event_callback = NULL;
```

Comment:
I believe we're trying to avoid new globals in favor of using the interpreter state.

Reply:
I think we're going to want a single callback per runtime, not per interpreter, so I moved this to `_PyRuntime`. Happy to change though if others disagree!